src/core/caniuse.js

```diff
@@ -0,0 +1,289 @@
+    Error [core/caniuse]: ${err.message}.
+    <br>Please check directly on <a href="${permalink}">${permalink}</a>.`;
+}
+
+const BROWSERS = { // browser name dictionary
+  chrome: "Chrome",
+  firefox: "Firefox",
+  ie: "IE",
+  edge: "Edge",
+  android: "Android",
+  safari: "Safari",
+  opera: "Opera",
+  bb: "Blackberry",
+  and_uc: "UC (Android)",
+  and_ff: "Firefox (Android)",
+};
+
+/**
+ * render the canIUse support table
+ * @param  {String} key       API name
+ * @param  {Object} stats     CanIUse API results
+ * @param  {Node} parent      where to render table
+ * @param  {Object} conf      respecConfig.caniuse
+ */
+function showData(key, stats, conf, parent) {
+  let browsers = conf.browsers;
+  if (conf.browsers === "ALL") {
+    browsers = Object.keys(stats);
+  }
+
+  // utils
+  const canIUseId = `caniuse-${key}`;
+  const permalink = `http://caniuse.com/#feat=${key}`;
+
+  const validBrowsers = browsers.filter(b => b in stats);
+  if (validBrowsers.length !== browsers.length) {
+    pub("warn", `Unsupported value(s) in \`conf.caniuse.browsers\`.
+    The following were ignored: \`[${browsers.filter(b => !(b in stats))}]\``);
+  }
+
+  // render the support table
+  const caniuse = hyperHTML`
+    <div class="caniuse" id="${canIUseId}">
+      <dt>Can I Use this API? (${key})</dt>
+      <div class="caniuse-stats">
+        ${validBrowsers.map(browser =>
+          addBrowser(browser, conf.versions, stats[browser])
+        )}
+        <a href="${permalink}" title="Get details at caniuse.com">
+          More info
+        </a>
+      </div>
+    </div>`;
+
+  const old = document.getElementById(canIUseId);
+  if (old) old.remove();
+  parent.parentNode.insertBefore(caniuse, parent.nextSibling);
+
+  /**
+   * add a browser and it's support to table
+   * @param {String} browser      name of browser (as in CanIUse API response)
+   * @param {Number} numVersions  number of old browser versions to show
+   * @param {Object} browserData  stats data from api response
+   */
+  function addBrowser(browser, numVersions, browserData) {
+    if (!browserData) return "";
+
+    const getSupport = version =>
+      browserData[version].split("#", 1)[0].trim();
+
+    const addBrowserVersion = version =>
+      `<div class="caniuse-cell ${getSupport(version)}">${version}</div>`;
+
+    const browserVersions = Object.keys(browserData)
+      .sort(semverCompare)
+      .slice(-(numVersions + 1)) // plus 1 current
+      .reverse();
+
+    return hyperHTML`
+    <div class="caniuse-browser">
+      <div class="${`caniuse-cell ${getSupport(browserVersions[0])}`}">
+        ${BROWSERS[browser] || browser} ${browserVersions[0]}
+      </div>
+      <div class="caniuse-col">
+        ${browserVersions.slice(1).map(addBrowserVersion)}
+      </div>
+    </div>`;
+  }
+}
+
+// promise based interface to IDB
+// supported methods: get, set, remove, clear, keys
+async function IDBCache(name, stores = ["caniuse"], version = 1) {
```

Comment:
so, let's move this to another module. And yeah, this would be super useful on its own! 

Also, what you have here is basically a class... so let's convert it to one:

```JS
// store databases by name here: 
const databases = new Map();

class IDBCache {
    constructor(name, stores = [], version = 1) {
        //  if databases has "name", use it. Otherwise, 
        // open new one and store it in databases.   
    }
    async ready(){ 
        //basically your getDatabase() function
    }
    async get(){
       const db = await this.ready();
    }
    async set(){} 
    async remove(){}
    async clear(){} 
    async keys(){}
}
```